velox/functions/prestosql/aggregates/AverageAggregate.cpp

```diff
@@ -313,6 +477,15 @@ bool registerAverageAggregate(const std::string& name) {
                            .argumentType("real")
                            .build());
 
+  signatures.push_back(
+      exec::AggregateFunctionSignatureBuilder()
+          .integerVariable("a_precision")
+          .integerVariable("a_scale")
+          .argumentType("DECIMAL(a_precision, a_scale)")
+          .intermediateType("ROW(DECIMAL(38, a_scale), bigint, bigint)")
```

Comment:
Do not forget to update Presto with this new intermediate type

Reply:
This is a first draft of the change. I am planning to update this to VARBINARY of (totalSum, count, overflow) to keep it same as JAVA.

If we decide to keep it as a ROW, then we have to define INT128 type to represent the total sum.

Reply:
To use `varbinary` as intermediate type, you probably need the support of solution 2 mentioned in #2869 to get the raw input type in an intermediate aggregation.  Let me know if you decide to do so.

Reply:
@Yuhta We wouldn't need the raw input type to compute the intermediate types. In case of decimal average, irrespective of input types, the sum is always int128.

Reply:
So for non-decimal input, you will keep the `row` type, and only change the intermediate type to `varbinary` when input is decimal?

Reply:
That is right.

```
 signatures.push_back(exec::AggregateFunctionSignatureBuilder()
                           .returnType("real")
                           .intermediateType("row(double,bigint)")
                           .argumentType("real")
                           .build());

 signatures.push_back(exec::AggregateFunctionSignatureBuilder()
                           .integerVariable("a_precision")
                           .integerVariable("a_scale")
                           .argumentType("DECIMAL(a_precision, a_scale)")
                           .intermediateType("VARBINARY")
                           .returnType("DECIMAL(a_precision, a_scale)")
                           .build());
```